src/Nethermind/Nethermind.Consensus.AuRa/AuRaSealValidator.cs

```diff
@@ -63,65 +68,83 @@ public bool ValidateParams(BlockHeader parent, BlockHeader header)
+                    if (!_validSealerStrategy.IsValidSealer(_validatorStore.GetValidators(), header.Beneficiary, step))
+                    {
+                        if (_logger.IsError) _logger.Error($"Block from incorrect proposer at block {header.ToString(BlockHeader.Format.FullHashAndNumber)}, step {step} from author {header.Beneficiary}.");
+                        return false;
+                    }
                 }
-            }
-            
-            // if (!ValidateEmptySteps())
-            // ReportBenign
-            ReportingValidator.TryReportSkipped(header, parent);
-            
-            // Report malice if the validator produced other sibling blocks in the same step.
-            if (_receivedSteps.ContainsSiblingOrInsert(header, _validatorStore.GetValidators().Length))
-            {
-                if (_logger.IsDebug) _logger.Debug($"Validator {header.Beneficiary} produced sibling blocks in the same step {header.AuRaStep} in block {header.Number}.");
-                ReportingValidator.ReportMalicious(header.Beneficiary, header.Number, Array.Empty<byte>(), IReportingValidator.MaliciousCause.SiblingBlocksInSameStep);
-            }
-            
-            if (header.Number >= _parameters.ValidateScoreTransition)
-            {
-                if (header.Difficulty >= AuraDifficultyCalculator.MaxDifficulty)
+
+                var currentStep = _stepCalculator.CurrentStep;
+
+                if (step > currentStep + rejectedStepDrift)
                 {
-                    if (_logger.IsError) _logger.Error($"Difficulty out of bounds for block {header.Number}, hash {header.Hash}, Max value {AuraDifficultyCalculator.MaxDifficulty}, but found {header.Difficulty}.");
+                    if (_logger.IsError) _logger.Error($"Block {header.Number}, hash {header.Hash} step {step} is from the future. Current step is {currentStep}.");
+                    ReportingValidator.ReportBenign(header.Beneficiary, header.Number, IReportingValidator.BenignCause.FutureBlock);
                     return false;
                 }
 
-                var expectedDifficulty = AuraDifficultyCalculator.CalculateDifficulty(parent.AuRaStep.Value, header.AuRaStep.Value, 0);
-                if (header.Difficulty != expectedDifficulty)
+                if (step > currentStep)
+                {
+                    const int blockTooEarlyWarningMillisecondThreshold = 1000;
+
+                    TimeSpan timeToStep = _stepCalculator.TimeToStep(step);
+                    if (timeToStep.TotalMilliseconds > blockTooEarlyWarningMillisecondThreshold)
+                    {
+                        if (_logger.IsWarn) _logger.Warn($"Block {header.Number}, hash {header.Hash} step {step} is {timeToStep:g} too early. Current step is {currentStep}.");
+                    }
+                }
+
+                // if (!ValidateEmptySteps())
```

Comment:
the commented out code should stay?

Reply:
Yes let it stay, its a TODO good reminder if we decide to implement empty steps